py3votecore/borda_at_large.py

```diff
@@ -0,0 +1,100 @@
+# Copyright (C) 2023, Leora Schmerler
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+from .abstract_classes import MultipleWinnerVotingSystem
+from .common_functions import matching_keys, unique_permutations
+import copy
+
+class BordaAtLarge(MultipleWinnerVotingSystem):
+    
+    def __init__(self, ballots: list, tie_breaker=None, required_winners: int=1):
+        """
+        The constructer accepts ballots of voters, a tie breaker if given and the number of required winners.
+        >>> BordaAtLarge([{ "count":3, "ballot":["A", "B", "C", "D"]},{ "count":2, "ballot":["D", "B", "A", "C"]},{ "count":2, "ballot":["D", "B", "C", "A"]}], required_winners = 2) # doctest:+ELLIPSIS
+        <...>
+        >>> BordaAtLarge([{ "count":3, "ballot":["A", "B", "C", "D"]},{ "count":2, "ballot":["D", "B", "A", "C"]},{ "count":2, "ballot":["D", "B", "C", "A"]}], tie_breaker=["A", "B", "C", "D"] ,required_winners = 2) # doctest:+ELLIPSIS
+        <...>
+        """
+        super(BordaAtLarge, self).__init__(ballots, tie_breaker=tie_breaker, required_winners=required_winners)
+
+    def calculate_results(self):
+        """
+        calculate_results accepts an instance of Borda, and is called from the constructor.
```

Comment:
Typically you would not state that it accepts anything if the only argument is the class instance itself.   This should be reworded to describe what the function actually does.